runtime/compiler/x/runtime/X86Codert.nasm

```diff
@@ -334,6 +334,72 @@ f2l_done:
 
 %endif ; TR_HOST_32BIT
 
+; crc32c
+;
+; Computes CRC32C checksum for a given buffer and size.
+;
+DECLARE_GLOBAL crc32c
+
+; rdi = crc > rax
+; rsi = bytes
+; rdx = off current rbx  - edx
+; rcx = end currently edx
+; tmp  = edi
+; tmp2 = r8
+align 16
+crc32c:
+        mov rax, rdi                        ; Copy current crc value into rax (result register)
+
+        begin512BitLoop:
+        lea rdi, [rdx+0x40]                 ;
+        cmp	rdi, rdx
+        jg begin64BitLoop                   ; Skip loop if less than 64 bytes in buffer
+
+        crc32 rax, qword [rsi+1*rdx+0x8]    ; 64-byte unrolled crc32c loop body
+        crc32 rax, qword [rsi+1*rdx+0x10]
+        crc32 rax, qword [rsi+1*rdx+0x18]
+        crc32 rax, qword [rsi+1*rdx+0x20]
+        crc32 rax, qword [rsi+1*rdx+0x28]
+        crc32 rax, qword [rsi+1*rdx+0x30]
+        crc32 rax, qword [rsi+1*rdx+0x38]
+        crc32 rax, qword [rsi+1*rdx+0x40]
+        add	rdx, 0x00000040                 ; Increase offset number of bytes written
+        jmp begin512BitLoop
+
+        begin64BitLoop:                     ;  crc 8 bytes at a time in loop
+        lea rdi, [rdx+0x8]
+        cmp	rdi, rcx
+        jg duff
+        crc32	rax, qword [rsi+1*rdx+0x8]
+        add	rdx, 0x00000008
+        jmp begin64BitLoop
+
+        duff:
```

Comment:
Rather than "duff", can you use something more descriptive like "crc32_residue"?